### PR TITLE
Sequential items

### PR DESCRIPTION
**Pull request by abdallahbeshi**

### CODE REVIEW

File: src/Mews.Fiscalization.Greece.Tests/IntegrationTests/AadeClientTests.cs [hunk 1]
```diff
@@ -67,12 +67,12 @@ public async Task ValidNegativeInvoiceWorks()
...
+                    revenueItems: SequentialEnumerableStartingWithOne.Create(new List<IndexedItem<NonNegativeRevenue>>
                     {
-                        new NonNegativeRevenue(new NonNegativeAmount(88.50m), new NonNegativeAmount(11.50m), TaxType.Vat13, RevenueType.Products),
-                        new NonNegativeRevenue(new NonNegativeAmount(88.50m), new NonNegativeAmount(11.50m), TaxType.Vat13, RevenueType.Services),
-                        new NonNegativeRevenue(new NonNegativeAmount(88.50m), new NonNegativeAmount(11.50m), TaxType.Vat13, RevenueType.Other)
-                    },
+                        new IndexedItem<NonNegativeRevenue>(1, new NonNegativeRevenue(new NonNegativeAmount(88.50m), new NonNegativeAmount(11.50m), TaxType.Vat13, RevenueType.Products)),
+                        new IndexedItem<NonNegativeRevenue>(2, new NonNegativeRevenue(new NonNegativeAmount(88.50m), new NonNegativeAmount(11.50m), TaxType.Vat13, RevenueType.Services)),
+                        new IndexedItem<NonNegativeRevenue>(3, new NonNegativeRevenue(new NonNegativeAmount(88.50m), new NonNegativeAmount(11.50m), TaxType.Vat13, RevenueType.Other))
+                    }),
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece.Tests/IntegrationTests/AadeClientTests.cs:75]: You can just use `SequentialEnumerableStartingWithOne.FromPreordered`. That will remove many changes in this PR.

File: src/Mews.Fiscalization.Greece/Model/SimplifiedInvoice.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@ public class SimplifiedInvoice : Invoice
         public SimplifiedInvoice(
             InvoiceHeader header,
             LocalCounterpart issuer,
-            IEnumerable<NonNegativeRevenue> revenueItems,
+            ISequentialEnumerable<NonNegativeRevenue> revenueItems,
```
[the remark this answers is not in the document]
Comment (abdallahbeshi) [reply src/Mews.Fiscalization.Greece/Model/SimplifiedInvoice.cs:11]: Before it was defaulting to 1, so i believe there is some validation on the gov side that it should be 1+.

[the remark this answers is not in the document]
Comment (KaliCZ) [reply src/Mews.Fiscalization.Greece/Model/SimplifiedInvoice.cs:11]: Then make iut be starting with 1, just make it more strongly type 😄

File: src/Mews.Fiscalization.Greece/Model/SimplifiedInvoice.cs [hunk 3]
```diff
@@ -8,7 +9,7 @@ public class SimplifiedInvoice : Invoice
         public SimplifiedInvoice(
             InvoiceHeader header,
             LocalCounterpart issuer,
-            IEnumerable<NonNegativeRevenue> revenueItems,
+            SequentialEnumerableStartingWithOne<NonNegativeRevenue> revenueItems,
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/SimplifiedInvoice.cs:12]: I meant in all the invoices (even the property.)

**Review by KaliCZ — APPROVED** [review]

**Pull request merged**